prog/weavewait/main.go

```diff
@@ -39,3 +61,43 @@ func checkErr(err error) {
 		os.Exit(1)
 	}
 }
+
+// Prepend the addresses to /etc/hosts, removing the docker address
+func updateHosts(iface *net.Interface) {
+	addrs, err := iface.Addrs()
+	checkErr(err)
+	if len(addrs) == 0 {
+		return
+	}
+	hostname, err := os.Hostname()
+	checkErr(err)
+	var oldHosts, newHosts bytes.Buffer
+	for _, addr := range addrs {
+		if addr, ok := addr.(*net.IPNet); ok {
+			fmt.Fprint(&newHosts, addr.IP, "\t", hostname, "\n")
+		}
+	}
+	f, err := os.OpenFile("/etc/hosts", os.O_RDWR, 0644)
+	checkErr(err)
+	defer f.Close()
+	_, err = oldHosts.ReadFrom(f)
+	checkErr(err)
+	if strings.Contains(oldHosts.String(), newHosts.String()) {
+		return
+	}
+	_, err = f.Seek(0, 0)
+	checkErr(err)
+	written, err := io.Copy(f, &newHosts)
+	checkErr(err)
+	// discard the first line, if it is the docker host
+	firstLine, err := oldHosts.ReadString('\n')
```

Comment:
correct. We remove the docker ip from /etc/hosts, so that it doesn't show up in `hostname -i`.